### PR TITLE
[processor/resourcedetection] Support Cloud Run Jobs

### PR DESCRIPTION
**Pull request by damemi**
**Description:** Adds support for Cloud Run Jobs using gcp-specific resource attributes

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** unit tests updated, downstream unit tests as well

**Documentation:** Semantic conventions updates: https://github.com/open-telemetry/semantic-conventions/blob/2246279243d743c6e073470f4e3551826f3115bc/specification/resource/semantic_conventions/cloud_provider/gcp/cloud_run.md

### CODE REVIEW

File: processor/resourcedetectionprocessor/internal/gcp/gcp.go [hunk 1]
```diff
@@ -92,6 +92,25 @@ func (d *detector) Detect(context.Context) (resource pcommon.Resource, schemaURL
 		if d.resourceAttributes.CloudRegion.Enabled {
 			b.add(conventions.AttributeCloudRegion, d.detector.FaaSCloudRegion)
 		}
+	case gcp.CloudRunJob:
+		if d.resourceAttributes.CloudPlatform.Enabled {
+			b.attrs.PutStr(conventions.AttributeCloudPlatform, conventions.AttributeCloudPlatformGCPCloudRun)
```
Comment (damemi) [inline processor/resourcedetectionprocessor/internal/gcp/gcp.go:97]: @jsuereth @dashpole @punya this is the only thing really blocking this, need to decide if Jobs fall under `AttributeCloudPlatformGCPCloudRun` or should be their own platform.

leaving this draft until that's resolved so it doesn't merge on accident

File: processor/resourcedetectionprocessor/internal/gcp/metadata.yaml [hunk 2]
```diff
@@ -48,4 +48,12 @@ resource_attributes:
...
     type: string
+    enabled: true
+  gcp.cloud_run.job.execution:
+    description: The Job execution name
+    type: string
+    enabled: true
+  gcp.cloud_run.job.task_index:
+    description: The Job execution task index
+    type: string
     enabled: true
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/internal/gcp/metadata.yaml:59]: With the new metatada approach, we should not add attributes enabled by default at the first iteration as it's considered a breaking change. They should be optional initially. If we want to enable them by default, this should be done as a follow-up change in couple of versions and this PR should have a warning like this https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/dockerstatsreceiver/metadata.yaml#L136-L137

Comment (dmitryax) [reply processor/resourcedetectionprocessor/internal/gcp/metadata.yaml:59]: Hm... I've just found that this is applied only for CloudRunJob which was not covered before, so it's probably good to have them enabled right away

**Review by dashpole — APPROVED** [review]

File: processor/resourcedetectionprocessor/internal/gcp/metadata.yaml (hunk 2 above)
Comment (damemi) [reply processor/resourcedetectionprocessor/internal/gcp/metadata.yaml:61]: @dmitryax is this good to resolve?

**Comment by atoulme** [comment]
Please resolve the conflict.

**Comment by damemi** [comment]
@atoulme updated, thanks

**Review by dmitryax — APPROVED** [review]

**Pull request merged**